src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -721,6 +725,9 @@ public void recover(Engine.RecoveryHandler recoveryHandler) throws EngineExcepti
     }
 
     public void failShard(String reason, Throwable e) {
+        if (engineClosed()) {
```

Comment:
I needed it for an Integration test I wrote, not sure if it makes sense though. See my comment above https://github.com/elastic/elasticsearch/pull/10610#issuecomment-93759161